src/nanoemoji/maximum_color.py

```diff
@@ -358,33 +367,38 @@ def _run(argv):
     if not FLAGS.destroy_non_color_glyphs:
         raise NotImplementedError("Retention of non-color glyphs not implemented yet")
 
-    input_font = Path(argv[1])
-    assert input_font.is_file()
-    font = ttLib.TTFont(input_font)
+    input_file = Path(argv[1]).resolve()  # we need a non-relative path
+    assert input_file.is_file()
+    font = ttLib.TTFont(input_file)
+    final_output = Path(config.load().output_file)
+    assert (
+        input_file.resolve() != (build_dir() / final_output).resolve()
+    ), "In == Out is bad"
 
     build_file = build_dir() / "build.ninja"
     build_dir().mkdir(parents=True, exist_ok=True)
 
-    # TODO flag control
     color_table = _vector_color_table(font)
 
     if gen_ninja():
         logging.info(f"Generating {build_file.relative_to(build_dir())}")
-        input_font = input_font.resolve()  # we need a non-relative path
         with open(build_file, "w") as f:
             nw = NinjaWriter(f)
-            _write_preamble(nw, input_font)
+            _write_preamble(nw)
+
+            wip_file = _keep_glyph_names(nw, input_file)
 
+            # generate the missing vector table
             if color_table == "COLR":
-                color_font, picosvg_files = _generate_svg_from_colr(
-                    nw, input_font, font
-                )
+                wip_file, picosvg_files = _generate_svg_from_colr(nw, wip_file, font)
             else:
-                color_font, picosvg_files = _generate_colr_from_svg(
-                    nw, input_font, font
-                )
+                wip_file, picosvg_files = _generate_colr_from_svg(nw, wip_file, font)
+
+            if FLAGS.bitmaps:
+                wip_file = _generate_cbdt(nw, input_file, font, wip_file, picosvg_files)
 
-            _generate_cbdt(nw, input_font, font, color_font, picosvg_files)
+            # until now we held glyph names stable, nuke 'em
+            _strip_glyph_names(nw, wip_file, final_output)
 
```

Comment:
glyph names may be useful for debugging so ideally maximum_color should respect the --keep_glyph_names flag (flase by default) like the main nanoemoji compiler?

Reply:
Oops, I forgot we had a flag for that. Good catch!